core/src/main/java/io/opencensus/tags/TagContexts.java

```diff
@@ -24,6 +24,11 @@
  *
  * <p>This class returns {@link TagContextBuilder builders} that can be used to create the
  * implementation-dependent {@link TagContext}s.
+ *
+ * <p>Implementations may have different constraints and are free to require their own subtypes of
+ * {@linkplain TagContext}. This means callers cannot assume the {@link #getCurrentTagContext()
```

Comment:
> are free to require their own subtypes of {@linkplain TagContext}.

Something like "are free to convert TagContexts to their own subtypes" might be clearer, since the implementation shouldn't throw an exception when given a different subtype.